app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt

```diff
@@ -165,4 +180,31 @@ class SettingsFragment : PreferenceFragmentCompat(), SettingsView {
         val switchLock = findPreference<SwitchPreference>("app_lock")
         switchLock?.isChecked = result == Authenticator.SUCCESS
     }
+
+    private fun removeSystemFromThemesIfNeeded() {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P) {
+            val pref = findPreference<ListPreference>("themes")
+            if (pref != null) {
+                val systemIndex = pref.findIndexOfValue("system")
+                if (systemIndex > 0) {
+                    var entries = pref.entries?.toMutableList()
+                    entries?.removeAt(systemIndex)
+                    var entryValues = pref.entryValues?.toMutableList()
+                    entryValues?.removeAt(systemIndex)
+                    if (entries != null && entryValues != null) {
+                        pref.entries = entries.toTypedArray()
+                        pref.entryValues = entryValues.toTypedArray()
+                    }
+                }
+            }
+        }
+    }
+
+    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences?, key: String?) {
```

Comment:
`ThemeHandler.setNightModeBaseOnTheme` will reload the activity duo to night mode change. `OnPreferenceChangeListener`
will trigger after change of preference but before write to shared preference. If we change the theme here, the  preference is never written to shared preference, because of the reset of the activity.

See here: https://stackoverflow.com/questions/51256723/onpreferencechangelistener-fires-before-change-is-made

Reply:
Can we do it after we persist the setting in the SettingsPresenter here then:
https://github.com/home-assistant/android/pull/780/files#diff-f13f037aed758b996ee8da1d6b628c06R78

Reply:
Yep this does work 👍